src/plugins/messageColors/index.tsx

```diff
@@ -0,0 +1,157 @@
+/*
+ * Vencord, a Discord client mod
+ * Copyright (c) 2024 Vendicated and contributors
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ */
+
+import "./styles.css";
+
+import { addChatBarButton } from "@api/ChatButtons";
+import { DataStore } from "@api/index";
+import { Devs } from "@utils/constants";
+import definePlugin from "@utils/types";
+import { React } from "@webpack/common";
+import type { Message } from "discord-types/general";
+
+import { ColorPickerChatButton } from "./ColorPicker";
+import { COLOR_PICKER_DATA_KEY, ColorType, regex, RenderType, savedColors, settings } from "./constants";
+
+export default definePlugin({
+    authors: [Devs.hen],
+    name: "MessageColors",
+    description: "Displays hex/rgb/hsl colors inside of a message",
+    settings,
+    patches: [{
+        find: "memoizeMessageProps:",
+        replacement: {
+            match: /function (\i)\((\i),(\i)\){return (.+?)(\i)}/,
+            replace: "function $1($2,$3){return $4$self.getColoredText($2,$3)}"
+        }
+    }],
+    async start() {
+        regex.push({
+            reg: settings.store.isHexRequired ?
+                /(#(?:[0-9a-fA-F]{3}){1,2})/g : /(#?(?:[0-9a-fA-F]{3}){1,2})/g,
+            type: ColorType.HEX
+        });
+
+        if (!settings.store.colorPicker) return;
+
+        addChatBarButton("vc-color-picker", ColorPickerChatButton);
+        let colors = await DataStore.get(COLOR_PICKER_DATA_KEY);
+        if (!colors) {
+            colors = [
+                1752220,
+                3066993,
+                3447003,
+                10181046,
+                15277667,
+                15844367,
+                15105570,
+                15158332,
+                9807270,
+                6323595,
+
+                1146986,
+                2067276,
+                2123412,
+                7419530,
+                11342935,
+                12745742,
+                11027200,
+                10038562,
+                9936031,
+                5533306
+            ];
+        }
+
+        savedColors.push(...colors);
+    },
+
+    getColoredText(message: Message, originalChildren: React.ReactElement[]) {
+        if (settings.store.renderType === RenderType.NONE) return originalChildren;
+        if (![0, 19].includes(message.type)) return originalChildren;
```

Comment:
magic numbers are really confusing and unmaintainable
```suggestion
        if (![MessageTypes.DEFAULT, MessageTypes.REPLY].includes(message.type)) return originalChildren;
```